compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs

```diff
@@ -403,14 +403,14 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         // that are not closures, then we type-check the closures. This is so
         // that we have more information about the types of arguments when we
         // type-check the functions. This isn't really the right way to do this.
+        self.select_obligations_where_possible(|_| {});
```

Comment:
What is this new call to select for? It may affect inference and also performance 🤔

Reply:
> What is this new call to select for? It may affect inference and also performance 🤔

sorry, commit the debug code, will be fix in next pr.